test/shopify_app/controller_concerns/login_protection_test.rb

```diff
@@ -100,6 +119,44 @@ class LoginProtectionControllerTest < ActionController::TestCase
     end
   end
 
+  test "#current_shopify_session loads offline session if token is signed with new secret" do
+    cookies.encrypted[ShopifyAPI::Auth::Oauth::SessionCookie::SESSION_COOKIE_NAME] = "cookie"
+    token = mock_jwt_token(@secret)
+    request.headers["HTTP_AUTHORIZATION"] = "Bearer #{token}"
+
+    ShopifyAPI::Utils::SessionUtils.expects(:load_current_session)
+      .with(
+        auth_header: "Bearer #{token}",
+        cookies: { ShopifyAPI::Auth::Oauth::SessionCookie::SESSION_COOKIE_NAME => "cookie" },
```

Comment:
We could probably remove the cookie part of this call since embedded apps won't rely on cookies.